File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <bits/stdc++.h>
using namespace std;

bool cmp(string a, string b) {
    return a + b > b + a;
}

string solution(vector<int> numbers) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    vector<string> nums;
    for (int num : numbers) {
        nums.push_back(to_string(num));
    }
    sort(nums.begin(), nums.end(), cmp);
    
    if(nums[0] == "0") return "0";
    
    string answer = "";
    for (string n : nums) {
        answer += n;
    }
    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String[] nums = new String[numbers.length];
        
        for (int i = 0; i < numbers.length; ++i) {
            nums[i] = String.valueOf(numbers[i]);
        }
        Arrays.sort(nums, (a, b) -> (b + a).compareTo(a + b));
        
        if (nums[0].equals("0")) return "0";
        
        StringBuilder answer = new StringBuilder();
        for (String num : nums) {
            answer.append(num);
        }        
        return answer.toString();
    }
}